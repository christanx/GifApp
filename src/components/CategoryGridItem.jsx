import React from 'react';
import PropTypes from 'prop-types';

const CategoryGridItem = ({ url, title }) => {
    return (
        <article className='card-grid-item animate__animated animate__fadeInUp'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </article>
    )
}

CategoryGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}

export default CategoryGridItem
