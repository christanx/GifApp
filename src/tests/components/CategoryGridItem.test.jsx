import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import CategoryGridItem from '../../components/CategoryGridItem';

describe('Testing CategoryGridItem', () => {
    const images = {
        id: 1,
        url: '/',
        title: 'Testing'
    }
    let wrapper = shallow(<CategoryGridItem { ...images } />);

    test('Component successful', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('El parrafo debe tener un titulo', () => {
        const title = wrapper.find('p').text().trim();
        expect(title).toBe(images.title)
    })
    
    test('La imagen debe tener las props src y alt que le corresponda', () => {
        const { src, alt } = wrapper.find('img').props();
        expect(src).toBe(images.url);
        expect(alt).toBe(images.title);
    })

    test('Debe tener la clase animate__fadeInUp', () => {
        const className = wrapper.find('article').prop('className');
        expect(className.includes('animate__fadeInUp')).toBe(true)
    })
    
    
    
})
