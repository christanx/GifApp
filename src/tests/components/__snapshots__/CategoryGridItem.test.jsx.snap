// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CategoryGridItem Component successful 1`] = `
<article
  className="card-grid-item animate__animated animate__fadeInUp"
>
  <img
    alt="Testing"
    src="/"
  />
  <p>
    Testing
  </p>
</article>
`;
